auth/auth.go

```diff
@@ -90,6 +90,14 @@ func (auth *Authenticator) GetUser(name string) (User, error) {
 
 // Looks up the information for a role.
 func (auth *Authenticator) GetRole(name string) (Role, error) {
+	role, err := auth.GetRoleIncDeleted(name)
+	if role != nil && role.IsDeleted() {
+		return nil, err
```

Comment:
Isn't err going to be nil here if it's soft-deleted?  I'd expected this method to return a NotFound error if it existed but was deleted, to match the previous behaviour.

Reply:
I was under the impression the previous behaviour would end up returning nil but I can check this.

Reply:
Yeah. The previous behaviour would return nil. Can discuss in sync up if need be